src/server/encode.rs

```diff
@@ -1,279 +1,101 @@
-        // Read bytes from body
-        let range = self.bytes_written..upper_bound;
-        let inner_poll_result = Pin::new(&mut self.res).poll_read(cx, &mut buf[range]);
-        let new_body_bytes_written = match inner_poll_result {
-            Poll::Ready(Ok(n)) => n,
-            Poll::Ready(Err(e)) => return Poll::Ready(Err(e)),
-            Poll::Pending => match self.bytes_written {
-                0 => return Poll::Pending,
-                n => return Poll::Ready(Ok(n)),
-            },
-        };
-        self.body_bytes_written += new_body_bytes_written;
-        self.bytes_written += new_body_bytes_written;
-
-        // Double check we did not read more body bytes than the total
-        // length of the body
-        debug_assert!(
-            self.body_bytes_written <= self.body_len,
-            "Too many bytes read. Expected: {}, read: {}",
-            self.body_len,
-            self.body_bytes_written
-        );
-
-        if self.body_len == self.body_bytes_written {
-            // If we've read the `len` number of bytes, end
-            self.dispatch(State::End, cx, buf)
-        } else if new_body_bytes_written == 0 {
-            // If we've reached unexpected EOF, end anyway
-            // TODO: do something?
-            self.dispatch(State::End, cx, buf)
-        } else {
-            // Else continue encoding
-            self.encode_fixed_body(cx, buf)
+        if self.response.header(DATE).is_none() {
+            let date = fmt_http_date(SystemTime::now());
+            self.response.insert_header(DATE, date);
         }
+        Ok(())
     }
 
-    /// Encode an AsyncBufRead using "chunked" framing. This is used for streams
-    /// whose length is not known up front.
-    fn encode_chunked_body(
-        &mut self,
-        cx: &mut Context<'_>,
-        buf: &mut [u8],
-    ) -> Poll<io::Result<usize>> {
-        let buf = &mut buf[self.bytes_written..];
-        match self.chunked.encode(&mut self.res, cx, buf) {
-            Poll::Ready(Ok(read)) => {
-                self.bytes_written += read;
-                match self.bytes_written {
-                    0 => self.dispatch(State::End, cx, buf),
-                    _ => Poll::Ready(Ok(self.bytes_written)),
-                }
+    /// Encode the headers to a buffer, the first time we poll.
+    fn compute_head(&mut self) -> io::Result<Cursor<Vec<u8>>> {
+        let mut head = Vec::with_capacity(100);
+        let reason = self.response.status().canonical_reason();
+        let status = self.response.status();
+        write!(head, "HTTP/1.1 {} {}\r\n", status, reason)?;
+
+        self.finalize_headers()?;
+        let mut headers = self.response.iter().collect::<Vec<_>>();
+        headers.sort_unstable_by_key(|(h, _)| h.as_str());
+        for (header, values) in headers {
+            for value in values.iter() {
+                eprintln!("{}: {}\r\n", header, value);
```

Comment:
this looks like debug code that should be removed

Reply:
🤫 this never happened